MonkeyWrench.Web.WebService/WebServices.asmx.cs

```diff
@@ -117,26 +129,27 @@ public void Logout (WebServiceLogin login)
 		}
 
 		[WebMethod]
-		public void CreateLanefile (WebServiceLogin login, int lane_id, string filename)
+		public void CreateLanefile (WebServiceLogin login, DBLane lane, string filename)
 		{
 			if (string.IsNullOrEmpty (filename))
 				throw new ArgumentException ("filename");
 
-			if (lane_id <= 0)
+			if (lane.id <= 0)
 				throw new ArgumentException ("lane_id");
 
 			using (DB db = new DB ()) {
-				VerifyUserInRole (db, login, Roles.Administrator);
-				db.Audit (login, "WebServices.CreateLaneFile (lane_id: {0}, filename: {1})", lane_id, filename);
+				VerifyUserInRoles (db, login, lane.required_roles, false);
```

Comment:
When I wrote it, I thought that these calls were only being made from inside EditLane, which requires you to have the permissions to get on the page at all. If this is an API call that can be access from outside of that scope, then yes, it won't work.

`[WebMethod]` does not really say "API Access" as much as `WebAPI`. I blame WebForms ;)